examples/dataset_configs.py

```diff
@@ -0,0 +1,137 @@
+    #'detection': None,
+    'segmentation': ('image', 'segmentedImage'),
+    'action_reco': ('video', 'audio', 'label')
+}
+
+
+class DatasetConfig:
+    def __init__(self,
+                 name,
+                 transform=transforms.ToTensor(),
+                 target_transform=None,
+                 train_arg=False,
+                 splits=None,
+                 split_arg_name='split',
+                 downloadable=True,
+                 special_args=None):
+
+        self.name = name
+        self.downloadable = downloadable
+        self.transform = transform
+        self.target_transform = target_transform
+        self.train_arg = train_arg
+        if train_arg:
+            self.splits = ('train', 'test')
+        elif splits is not None:
+            self.splits = splits
+        else:
+            self.splits = None
+        self.split_arg_name = split_arg_name
+        self.special_args = special_args
+
+    def __len__(self):
+        if self.splits is not None:
+            return len(self.splits)
+        else:
+            return 0
+
+    def gen_kwargs(self, root_dir):
+        kwargs = {}
+        kwargs['root'] = root_dir
+        if self.downloadable:
+            kwargs['download'] = True
+        if self.transform:
+            kwargs['transform'] = self.transform
+        if self.target_transform:
+            kwargs['target_transform'] = self.target_transform
+        if self.special_args is not None:
+            kwargs.update(self.special_args)
+
+        dataset_names = []
+        if self.__len__() > 0:
+            kwargs_list = []
+            if self.train_arg:
+                kwargs['train'] = True
+                kwargs_list.append(kwargs.copy())
+                dataset_names.append(f'{self.name}_train')
+                kwargs['train'] = False
+                kwargs_list.append(kwargs.copy())
+                dataset_names.append(f'{self.name}_test')
+            else:
+                for split in self.splits:
+                    kwargs[self.split_arg_name] = split
+                    kwargs_list.append(kwargs.copy())
+            kwargs = kwargs_list
+
+        return dataset_names, kwargs
+
+# TorchVision Datasets
+# TODO add VOC, COCO
+mnist = DatasetConfig('MNIST', train_arg=True)
+cifar10 = DatasetConfig('CIFAR10', train_arg=True)
+fashion_mnist = DatasetConfig('FashionMNIST', train_arg=True)
+
+resize = transforms.Compose([
+                    transforms.Resize([224, 224]),
+                    transforms.ToTensor(),])
+image_folder = DatasetConfig('ImageFolder',
+                             transform=resize,
+                             splits=None,
+                             downloadable=False)
+voc_seg = DatasetConfig('VOCSegmentation',
+                        transform=resize,
+                        target_transform=resize,
+                        split_arg_name='image_set',
+                        splits=['train', 'val', 'trainval'])
+
+def voc_target_transform(target):
```

Comment:
I'm wondering if transforms specific to particular datasets should live in a different module. This doesn't seem to belong in a module called `configs.py`. 

Also, some documentation here would be helpful, since I wouldn't assume knowledge of a particular datasets content / the necessary transforms required to make the data useable.

Reply:
Yes it should actually. Will move it over